src/transformState.js

```diff
@@ -4,8 +4,29 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+
 function transformState(state, actions) {
-  // write code here
+  const trans = state;
+
+  for (const a in actions) {
+    switch (actions[a].type) {
+      case 'addProperties':
+        Object.assign(trans, actions[a].extraData);
+        break;
+      case 'removeProperties':
+        for (const k of actions[a].keysToRemove) {
```

Comment:
Same here and below. You can also make a [destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) to get all the properties you need from actions[action]